### PR TITLE
Add sample code in PHP, Python and JavaScript for API:Filearchive

### PR DESCRIPTION
**Pull request by lizwkariuki58**

### CODE REVIEW

**Comment by brendajerop** [comment]
@lizwkariuki58 Thank you for working on this!
A few initial comments:
* Since API:Filearchive is a GET request, the code should be auto-generated. How you do that is by adding its JSON schema to [modules.json](https://github.com/wikimedia/mediawiki-api-demos/blob/master/modules.json), then run `python autogenerator.py` to autogenerate the code. Please see the last statement in [the contributing guidelines](https://github.com/wikimedia/mediawiki-api-demos/blob/master/contributing.md).
* Please provide a link to the sandbox page for this module.
* In your commit, make sure you add API:Filearchive to the readme files of Python, PHP and JavaScript folders.

Feel free to ask if any of this is not clear :)

**Comment by lizwkariuki58** [comment]
Hey
I have done the corrections you have suggested above.

**Comment by lizwkariuki58** [comment]
But I am getting some merge conflicts. Should I merge my feature branch into my master first?

**Comment by brendajerop** [comment]
@lizwkariuki58 Yes, the merge conflicts should be resolved :)

**Comment by lizwkariuki58** [comment]
Hey
I resolved the merge conflicts

**Comment by lizwkariuki58** [comment]
The URL for my sandbox is: https://www.mediawiki.org/wiki/User:Liz_Kariuki/Sandbox/API:Filearchive

File: javascript/README.md [hunk 1]
```diff
@@ -57,20 +57,19 @@ Code snippets in Javascript demonstrating how to use various modules of the [Med
...
 * [API:Embeddedin](https://www.mediawiki.org/wiki/API:Embeddedin)
   * [get_embedded_pages.js](get_embedded_pages.js): get all page(s) that embed a page
 * [API:Mergehistory](https://www.mediawiki.org/wiki/API:Mergehistory)
   * [mergehistory.js](mergehistory.js): Merge the page revisions of Oldpage
     dating up to 2015-12-31T04:37:41Z into Newpage
-* [API:Contributors](https://www.mediawiki.org/wiki/API:Contributors)
-  * [get_contributors.js](get_contributors.js): get request to list all logged-in contributors and count of anonymous contributors to a page
-* [API:Stashedit](https://www.mediawiki.org/wiki/API:Stashedit)
-  * [stashedit.js](stashedit.js): Prepare an edit in shared cache
+* [API:FileArchive](https://www.mediawiki.org/wiki/User:Liz_Kariuki/Sandbox/API:Filearchive)
```
Comment (brendajerop) [inline javascript/README.md]: This should link to the page in the main namespace, i.e. `https://www.mediawiki.org/wiki/API:Filearchive`

File: modules.json [hunk 2]
```diff
@@ -570,28 +570,18 @@
...
+        "filename": "filearchive",
+        "docstring": "Enumerate 3 deleted files from filearchive table in descending order.",
         "endpoint": "https://en.wikipedia.org/w/api.php",
         "params": {
             "action": "query",
-            "list": "alldeletedrevisions",
-            "adruser": "Mahesh",
-			"adrprop": "ids|user|comment",
-			"format": "json"
+			"format": "json",
```
Comment (brendajerop) [inline modules.json]: The indentation of the schema is inconsistent. Check if you are mixing tabs and spaces.

**Comment by lizwkariuki58** [comment]
Hey
Made the corrections

File: javascript/README.md [hunk 3]
```diff
@@ -71,8 +71,10 @@ Code snippets in Javascript demonstrating how to use various modules of the [Med
   * [get_contributors.js](get_contributors.js): get request to list all logged-in contributors and count of anonymous contributors to a page
 * [API:Stashedit](https://www.mediawiki.org/wiki/API:Stashedit)
   * [stashedit.js](stashedit.js): Prepare an edit in shared cache
+* [API:Filearchive](https://www.mediawiki.org/wiki/API:Filearchive)
```
Comment (brendajerop) [inline javascript/README.md]: @lizwkariuki58 I'm wondering if `API:Filearchive` is suitable under `Accounts and users`. Maybe under page operations?

**Comment by brendajerop** [comment]
I've seen that you have sent a commit to resolve merge conflicts, however I don't think you rebased your changes on top of the latest changes in master because there's still a conflict :/. To rebase, use `git pull --rebase upstream master`.

**Comment by brendajerop** [comment]
@lizwkariuki58 I've left some feedback on the discussion page of your sandbox :)

**Pull request merged**